### ISSUE TITLE
⚖️ Address Search on Map

### ISSUE THREAD

**Issue by vogelino** [vogelino/data-2-resilience#24]
Searching for an address should be geocoded and the first result should be displayed.
The distance between the address found and all stations should be calculated (eg. using turf) and the closest station(s) should be suggested. 
When clicking on the address, the map should be zoomed to this place and a temporary marker should be displayed
When a station is selected, it should be toggled from the left sidebar, as if it had been clicked and the map should zoom to it (eventually it should only zoom to it).
On the heat atlas page, stations are not suggested, as they aren't displayed.

PREVIEW: https://data-2-resilience.vercel.app/

**Comment by alsino** [comment]
Dortmund Geocode API:

"...sie können die BKG API nutzen (s. Anhang). Unsere Adresse zur API lautet https://geoweb1.digistadtdo.de/doris_gdi/geocoder

Beispiel Adressen Suche: https://geoweb1.digistadtdo.de/doris_gdi/geocoder/geosearch?query=Meuselwitzstra%C3%9Fe%201%20Dortmund&outputformat=JSON
Beispiel Adressen Suche mit Koordinatensystem: https://geoweb1.digistadtdo.de/doris_gdi/geocoder/geosearch?query=Meuselwitzstra%C3%9Fe%201%20Dortmund&srsName=EPSG:25832&outputformat=JSON
Beispiel Reverse Suche: https://geoweb1.digistadtdo.de/doris_gdi/geocoder/geosearch?lon=7.527944&lat=51.553257&distance=100&outputformat=JSON

Über diesen Dienst könne sie aber nur Adressen in der Nähe von Dortmund finden. Die BKG Suche ist entsprechend eingegrenzt"

**Comment by vogelino** [comment]
Very helpful thx!

**Comment by vogelino** [comment]
@alsino As noted in our last group meeting, there were a few issues related to search. Those were fixed:
- ✅ Selecting a search suggestion via click does now work #40 
- ✅ Subsequent searches do directly open the suggestions #30 

Let me know if you notice anything else! Cheers!

**Comment by alsino** [comment]
@vogelino Yeah, veeeeeery nice! Really like the user experience now 🙌

One question: Do you think it is worth considering to preserve the map search results after languages have been switched? 

Scenario: 
I am set to DE and search for a place. Then I switch to EN and still wannt to see the marker on the map. Right ow the marker disappears.

But maybe this is not very likely and thus overkill?

**Comment by vogelino** [comment]
@alsino good catch. The map search query is not part of the url state (This would require additional logic to make sure that the search suggestions are not triggered on page load, which is quite tricky and has other implications). 

Because changing the language means changing the navigation, the local state is reset. There might be a way to solve that (maybe with an external store), but I'm not 100% sure how.

 I think we should move on for know and I will add an issue for this. What do you think? Cheers!

**Comment by alsino** [comment]
Yes, you are right! I think this is not super urgent as it is not a very common scenario. 
Let's not get entangled in this for now.

**Comment by vogelino** [comment]
> Let's not get entangled in this for now.

Al right! Let's do this later then. Chévere
